qiime_studio/api/v1.py

```diff
@@ -50,4 +57,61 @@ def api_workflows(plugin_name):
             {'name': name, 'type': repr(type_[0])}
             for name, type_ in value.signature.outputs.items()
         ]
-    return jsonify({"workflows": workflows_dict})
+    return jsonify({'workflows': workflows_dict})
+
+
+@v1.route('/artifacts', methods=['GET'])
+def api_artifacts():
+    artifact_paths = glob.glob(os.path.join(os.getcwd(), '*.qzf'))
+    artifacts = [
+        {
+            'name': os.path.splitext(os.path.split(path)[1])[0],
+            'uuid': str(Artifact.load(path).uuid),
+            'type': str(Artifact.load(path).type),
+            'path': path,
+            'uri': 'artifacts/%s' % (os.path.splitext(
+                                        os.path.split(path)[1])[0])
+        }
+        for path in artifact_paths
+    ]
+    return jsonify({'artifacts': artifacts})
+
+
+@v1.route('/artifacts/<name>', methods=['DELETE'])
+def delete_artifact(name):
+    result = {
+        'success': True
+    }
+    artifact_json = request.get_json()['artifact']
+    artifact = Artifact.load(artifact_json['path'])
+    if str(artifact.uuid) == artifact_json['uuid']:
+        try:
+            os.remove(artifact_json['path'])
+        except OSError:
+            result['success'] = False
+
+    return jsonify(result)
+
+
+@v1.route('/artifacts/<plugin_name>/<workflow_name>/<input_name>', methods=['GET'])
+def api_input_artifacts(plugin_name, workflow_name, input_name):
+    plugin = PLUGIN_MANAGER.plugins[plugin_name]
+    workflow = plugin.workflows[workflow_name]
+    input_type = workflow.signature.inputs[input_name][0]
+    input_artifacts = []
+    artifact_paths = glob.glob(os.path.join(os.getcwd(), '*.qzf'))
+    artifacts = [
+        {
+            'name': os.path.splitext(os.path.split(path)[1])[0],
+            'uuid': str(Artifact.load(path).uuid),
+            'type': str(Artifact.load(path).type),
+            'path': path,
+            'uri': 'artifacts/%s' % (os.path.splitext(
+                                        os.path.split(path)[1])[0])
+        }
+        for path in artifact_paths
+    ]
+    for artifact in artifacts:
+        if Artifact.load(artifact['path']).type <= input_type:
+            input_artifacts.append(artifact)
```

Comment:
I might refactor this so that `Artifact.load` is only called once:

```python
for artifact in map(Artifact.load, artifact_paths):
    if artifact.type <= input_type:
         input_artifacts.append({
               <your json>
         })
```